hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/crypto/OpensslCipher.c

```diff
@@ -207,10 +233,20 @@ JNIEXPORT void JNICALL Java_org_apache_hadoop_crypto_OpensslCipher_initIDs
   LOAD_DYNAMIC_SYMBOL(__dlsym_EVP_CIPHER_CTX_test_flags,  \
                       dlsym_EVP_CIPHER_CTX_test_flags, env,  \
                       openssl, "EVP_CIPHER_CTX_test_flags");
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
```

Comment:
these are all static...the probes need to be compiled for openssl 2 vs 3. do we need this, or can it just look for either sets of symbols and be happy?

Reply:
Not quite sure - let me check.

Reply:
I need some more information here:

1. With "look for either sets of symbols" do you mean moving the conditional symbol loading from compile time to runtime?
2. What contract do we have for native code regarding OpenSSL ABIs?

The PR delivers the following contract:
* If compiled against OpenSSL X.Y API, it will only work at runtime if OpenSSL X.Y ABI is available.

To my knowledge, this is not a regression, looking at the other conditional symbol loads and their usage.